152_data_reading_program/sensor_reader.py

```diff
@@ -0,0 +1,85 @@
+# This script is used to gather data from the SCD-30 sensor attached to
+# the MCP-2221. This script is influenced by Carter Nelson and
+# Bryan Siepert's scripts written for Adafruit Industries in 2020 and 2021
+# It enhances these scripts by calling the linux export statement directly,
+# By presenting the data much more cleanly
+# by having a discrete function to reset the SCD and calling this function
+# when necessary when the SCD-30 crashes (which happens sometimes)
+# and also by recovering from failure to read from the SCD-30
+# by trying again next time instead of just crashing altogether.
+# Unfortunately, there is still a problem where the sensor stops working
+# eventually and must be unplugged from the computer and plugged back in
+# again to make it work.
+# @author Greg Ross, version 1, October 29,2021
+
+# According to Carter'script, the following commented line causes
+# prevents an issue with the reset on MCP-2221
+# pylint: disable=protected-access
+
+import sys # For force quit.
+import os # To do the system call to set export
+# Set the export statment command with a system call
+# in case it is not already set.
+# If this system call is not made, import board will fail
+os.system("export BLINKA_MCP2221=1")
+import time # For sleep function
+import adafruit_scd30 # Adafruit's Library for the SCD-30
+import busio # For I2C
+import board # For MCP-2221
+
+# From Carter's Script, it is recommended to start with a frequency
+# of 50 Khz due to "Clock Stretching.
+i2c = busio.I2C(board.SCL, board.SDA, frequency=50000)
+scd = adafruit_scd30.SCD30(i2c)
+
+# scd.reset() apparently isn't necessary as long as the sensor is not in an
+# error state to start with. scd.reset() is alternative to the
+# following which is supposed to work better for MCP-2221
+def reset_scd():
+    #Check if this script is run on MCP-2221 (Carter's )
+    if hasattr(i2c, "_i2c"):
+        if hasattr(i2c._i2c, "_mcp2221"):
```

Comment:
```suggestion
    if hasattr(i2c, "_i2c") and hasattr(i2c._i2c, "_mcp2221"):
```

The boolean operators are lazy, so the second won't be evaluated if the first fails.